vsintegration/src/FSharp.ProjectSystem.FSharp/Project.fs

```diff
@@ -1620,14 +1663,16 @@ namespace rec Microsoft.VisualStudio.FSharp.ProjectSystem
                         o :?> System.String
                     else
                         null : System.String
+
             let OnActiveProjectCfgChange(pIVsHierarchy) =
-                if GetCaption(pIVsHierarchy) = GetCaption(projNode.InteropSafeIVsHierarchy) then
+                if GetCaption(pIVsHierarchy) = GetCaption(projNode.InteropSafeIVsHierarchy) && batchState <> BatchDone then
                     projNode.SetProjectFileDirty(projNode.IsProjectFileDirty)
-                    projNode.ComputeSourcesAndFlags() // REVIEW: It looks like ComputeSourcesAndFlags is called twice. Once on this line and then again because it is added to 'queuedWork' below.
-                    match queuedWork with
-                    | Some(l) -> queuedWork <- Some( projNode :: l )
-                    | None -> ()
+                    projNode.ComputeSourcesAndFlags()
+                    
+                    if batchState = BatchWaiting then
```

Comment:
This does not look threadsafe. 

Is there some mechanism in place that ensures this can never race, or should we use 
Interlocked Compex or a lock?